### ISSUE TITLE
Rule S2068: support url userinfo component

### ISSUE THREAD

**Issue by pierre-loup-tristant-sonarsource** [SonarSource/sonar-dotnet#2865]
[RSPEC-2068](https://jira.sonarsource.com/browse/RSPEC-2068)

The URL user info component can contain a hardcoded password:

```csharp
string url1 = "scheme://user:azerty123@domain.com"; // Sensitive
string url2 = "scheme://user:@domain.com"; // Compliant
string url3 = "scheme://user@domain.com:80"; // Compliant
string url4 = "scheme://user@domain.com"; // Compliant
string url5 = "scheme://domain.com/user:azerty123"; // Compliant
```

Exception: no issue should be raised if user and password part of the userinfo component are the same:

```
string url1 = "scheme://admin:admin@domain.com"; // Compliant
```
